gdsfactory/export/to_kfactory.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+
+import kfactory as kf
+
+import gdsfactory as gf
+
+
+def import_gds(gdspath: str | Path) -> kf.KCell:
+    """Reads a GDS file and returns a kfactory component.
+
+    Args:
+        gdspath: path to GDS file.
+    """
+    read_options = kf.kcell.load_layout_options()
+    read_options.cell_conflict_resolution = (
+        kf.kdb.LoadLayoutOptions.CellConflictResolution.RenameCell
+    )
+    top_cells = set(kf.kcl.top_cells())
+    kf.kcl.read(gdspath, read_options, test_merge=True)
+    new_top_cells = set(kf.kcl.top_cells()) - top_cells
+    if len(new_top_cells) != 1:
+        raise ValueError(f"Expected 1 new top cell, got {len(new_top_cells)}")
+    return kf.kcl[new_top_cells.pop().name]
+
+
+def to_kfactory(component) -> kf.KCell:
+    """Converts a gdsfactory component to a kfactory component."""
+    c0 = component.write_gds(with_metadata=True)
+    c = import_gds(c0)
+
+    for k, v in dict(component.info).items():
+        c.info[k] = v
+    for k, v in dict(component.settings).items():
+        if isinstance(v, str | int | float | bool):
+            c.info[k] = v
```

Comment:
You could also write the settings with

`c._settings = kf.kcell.KCellSettings(**component.settings.model_dump())` (maybe needs filtering for the types)